frontend/packages/kubevirt-plugin/src/components/modals/cdrom-vm-modal/index.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import { getNamespace } from '@console/shared';
+import { Firehose } from '@console/internal/components/utils';
+import { createModalLauncher, ModalComponentProps } from '@console/internal/components/factory';
+import { PersistentVolumeClaimModel, StorageClassModel } from '@console/internal/models';
+import { VMKind } from '../../../types';
+import { CDRomModal } from './cdrom-modal';
+
+const CDRomModalFirehose: React.FC<CDRomModalFirehoseProps> = (props) => {
+  const { vm } = props;
+
+  const resources = [
+    {
+      kind: StorageClassModel.kind,
+      isList: true,
+      prop: 'storageClasses',
+    },
+    {
+      kind: PersistentVolumeClaimModel.kind,
+      isList: true,
+      namespace: getNamespace(vm),
+      prop: 'persistentVolumeClaims',
+    },
+  ];
+
+  return (
+    <Firehose resources={resources}>
+      <CDRomModal {...props} />
+    </Firehose>
+  );
+};
+
+type CDRomModalFirehoseProps = ModalComponentProps & {
+  vm: VMKind;
+  dataVolume?: any;
```

Comment:
nit: would be better to have `DataVolumeKind` type here, but this is not a blocker.

Reply:
unused, removed it